src/date.ml

```diff
@@ -75,8 +75,16 @@ let hash = Utils.Int.hash
 
 (* Constructors. *)
 
-let lt (d1 : int * int * int) (d2 : int * int * int) =
-  Pervasives.compare d1 d2 < 0
+let lt ((y1, m1, d1) : int * int * int) ((y2, m2, d2) : int * int * int) =
+  let cmp = match compare y1 y2 with
+  | 0 -> begin
+      match compare m1 m2 with
+      | 0 -> compare d1 d2
+      | i -> i
+    end
+  | i -> i
+  in
+  cmp < 0
```

Comment:
Shorter and nicer:
```ocaml
let cmp_int (a: int) b = compare a b
let (<?>) c (ord,x,y) =
  if c = 0 then ord x y else c
let lt ((y1, m1, d1) : int * int * int) ((y2, m2, d2) : int * int * int) =
  cmp_int y1 y2 <?> (cmp_int, m1, m2) <?> (cmp_int, d1, d2)
```

Reply:
Actually, since you're on it, you might as well introduce properly efficient comparisons everywhere :)

Reply:
The infix operator indeed makes things very readable here. Thanks!

For the other comparisons i believe the ones i saw so far were already using the integer compare from the `Int` module inside utils (`let compare (a: int) b = compare a b`) 

I'll take a second pass to find places i missed. I will be on vacation for a week so i won't be able to update the pull request for a while.